### ISSUE TITLE
getpass.getpass() fatally broken on Windows

### ISSUE THREAD

**Issue by JamesTheAwesomeDude** [pypy/pypy#4881]
```python
# Python 3.10.13 (fc59e61cfbff, Jan 14 2024, 16:27:58)
# [PyPy 7.3.15 with MSC v.1929 64 bit (AMD64)] on win32
# Type "help", "copyright", "credits" or "license" for more information.
import getpass

getpass.getpass('u> ')
# Traceback (most recent call last):
#   File "<stdin>", line 1, in <module>
#   File "C:\Users\AdministratorJames\OneDrive - Arcfield\Programs\PyPy310\Lib\getpass.py", line 103, in win_getpass
#     msvcrt.putwch(c)
#   File "C:\Users\AdministratorJames\OneDrive - Arcfield\Programs\PyPy310\Lib\msvcrt.py", line 112, in putwch
#     _lib._putwch(ord(ch))
# TypeError: initializer for ctype 'wchar_t' must be a unicode string of length 1, not int

getpass.getpass(b'b> ')
# Traceback (most recent call last):
#   File "<stdin>", line 1, in <module>
#   File "C:\Users\AdministratorJames\OneDrive - Arcfield\Programs\PyPy310\Lib\getpass.py", line 103, in win_getpass
#     msvcrt.putwch(c)
#   File "C:\Users\AdministratorJames\OneDrive - Arcfield\Programs\PyPy310\Lib\msvcrt.py", line 112, in putwch
#     _lib._putwch(ord(ch))
# TypeError: ord() expected string of length 1, but int found
```

Maybe related to #2999

**Comment by JamesTheAwesomeDude** [comment]
It's worth noting that this renders `twine upload` completely unusable

https://github.com/pypa/twine/blob/4.0.2/twine/auth.py#L93

**Comment by mattip** [comment]
Fixed in 137e067a8. You can make the change to the pure-python msvcrt.py locally.

**Comment by mattip** [comment]
Thanks for the report. Closing, please reopen or open a new issue as needed.
